src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RedundantThisCheck.java

```diff
@@ -0,0 +1,364 @@
+ *   }
+ *
+ *   public int methodOne() {
+ *     methodThree();
+ *     return this.x++; // OK, fields are not checked
+ *   }
+ *
+ *   public int methodTwo() {
+ *     this.methodThree(); // violation, 'this' is not necessary here
+ *   }
+ *
+ *   public int methodThree() {}
+ * }
+ * </pre>
+ *
+ * @since 8.32
+ */
+// -@cs[ClassDataAbstractionCoupling] This check requires to work with and identify many frames.
+public class RedundantThisCheck extends AbstractFrameCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_METHOD = "redundant.this.method";
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_VARIABLE = "redundant.this.variable";
+
+    /** Control whether to check references to fields. */
+    private boolean checkFields = true;
+
+    /** Control whether to check references to methods. */
+    private boolean checkMethods = true;
+
+    /**
+     * Setter to control whether to check references to fields.
+     * @param checkFields should we check methods usage or not.
+     */
+    public void setCheckFields(boolean checkFields) {
+        this.checkFields = checkFields;
+    }
+
+    /**
+     * Setter to control whether to check references to methods.
+     * @param checkMethods should we check methods usage or not.
+     */
+    public void setCheckMethods(boolean checkMethods) {
+        this.checkMethods = checkMethods;
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {
+            TokenTypes.CLASS_DEF,
+            TokenTypes.INTERFACE_DEF,
+            TokenTypes.ENUM_DEF,
+            TokenTypes.ANNOTATION_DEF,
+            TokenTypes.CTOR_DEF,
+            TokenTypes.METHOD_DEF,
+            TokenTypes.LITERAL_FOR,
+            TokenTypes.SLIST,
+            TokenTypes.LITERAL_TRY,
+            TokenTypes.LITERAL_CATCH,
+            TokenTypes.INSTANCE_INIT,
+            TokenTypes.LITERAL_THIS,
+        };
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        switch (ast.getType()) {
+            case TokenTypes.LITERAL_THIS :
+                processLiteralThis(ast);
+                break;
+            case TokenTypes.CLASS_DEF :
+            case TokenTypes.INTERFACE_DEF :
+            case TokenTypes.ENUM_DEF :
+            case TokenTypes.ANNOTATION_DEF :
+            case TokenTypes.SLIST :
+            case TokenTypes.LITERAL_TRY :
+            case TokenTypes.LITERAL_CATCH :
+            case TokenTypes.INSTANCE_INIT :
+            case TokenTypes.METHOD_DEF :
+            case TokenTypes.CTOR_DEF :
+            case TokenTypes.LITERAL_FOR :
```

Comment:
no need for whitespace before colon.

I was so surprised to see such syntax that it inspired me to create issue for it https://github.com/checkstyle/checkstyle/issues/8045

Reply:
Done.
Do you want me to correct this in `RequireThisCheck` as well? may be on a separate PR?

Reply:
I think RequireThis can be corrected as well

Reply:
Done.